pool/pool.go

```diff
@@ -860,7 +863,7 @@ func (p *Pool) updateInnerNodesHealth(ctx context.Context, i int, bufferWeights
 	pool := p.innerPools[i]
 	options := p.rebalanceParams
 
-	healthyChanged := atomic.NewBool(false)
+	var healthyChanged atomic.Bool
```

Comment:
I'd keep it a pointer.

Reply:
why?

Reply:
dont mind, wanted to hear some thoughts, nothing more

Reply:
Updated